sqlite/migrate.go

```diff
@@ -183,11 +184,84 @@ func GetLastMigrationVersion(db *sql.DB) (version uint, migrationTableExists boo
 
 	var lastMigration uint64 = 0
 	if migrationTableExists {
-		row = db.QueryRow("SELECT version FROM status_go_schema_migrations")
+		row = db.QueryRow(fmt.Sprintf("SELECT version FROM %s", migrationTableName))
 		err = row.Scan(&lastMigration)
 		if err != nil && err != sql.ErrNoRows {
 			return 0, true, err
 		}
 	}
 	return uint(lastMigration), migrationTableExists, nil
 }
+
+// Used to migrate from one migration table to another, ensuring the table reflects the correct migration state.
+// Ensures migrationTableName exists and records the latest version from assetNames, capped at maxVersion.
+// Intended for migration table transitions and version synchronization.
+func UpdateMigrationTableVersion(db *sql.DB, migrationTableName string, assetNames []string, maxVersion uint) error {
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err != nil {
+			rollbackErr := tx.Rollback()
+			if rollbackErr != nil {
+				err = fmt.Errorf("failed to rollback transaction: %w; original error: %v", rollbackErr, err)
+			}
+		}
+	}()
+
+	row := tx.QueryRow("SELECT exists(SELECT name FROM sqlite_master WHERE type='table' AND name=?)", migrationTableName)
+	exists := false
+	err = row.Scan(&exists)
+	if err != nil && err != sql.ErrNoRows {
+		return err
+	}
+
+	if !exists {
```

Comment:
Is it ever possible that the migration table already exists, but we still need to update the last applied version in it? 🤔